src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1522,14 +1522,17 @@ private static ReadOnlySpan<char> SetFromProperty(string capname, bool invert, s
 
 #if DEBUG
         public static readonly char[] Hex = new char[] { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
-        public static readonly string[] Categories = new string[] {"Lu", "Ll", "Lt", "Lm", "Lo", InternalRegexIgnoreCase,
-                                                                     "Mn", "Mc", "Me",
-                                                                     "Nd", "Nl", "No",
-                                                                     "Zs", "Zl", "Zp",
-                                                                     "Cc", "Cf", "Cs", "Co",
-                                                                     "Pc", "Pd", "Ps", "Pe", "Pi", "Pf", "Po",
-                                                                     "Sm", "Sc", "Sk", "So",
-                                                                     "Cn" };
+        public static readonly string[] _categoryIdToName = PopulateCategoryIdToName();
+
+        private static string[] PopulateCategoryIdToName()
+        {
+            // Populate category reverse lookup used for diagnostic output
+
+            var temp = new List<KeyValuePair<string, string>>(s_definedCategories);
+            temp.RemoveAll(kvp => kvp.Value.Length != 1);
```

Comment:
I don't understand how this maps to what was there previously.  You're keeping only categories that are a single letter?

Reply:
Yes. This table is only ever used to look up a single character. For example, given ` { "Lu", "\u0001" }, ` in the regular table, it wants to look up `\u0001` and get "Lu".  The original table has some longer values, eg., `{ "C", "\u0000\u000F\u0010\u001E\u0012\u0011\u0000" },`. These are just concatenations of regular entries - in this case "C" is equivalent to "Cc", "Cf", etc and the dump code just looks up those individually. The dump does not attempt to recognize and collapse into shorter sequences, eg., collapsing "Cc", "Cf" etc into "C". At the point of dump it doesn't know what the original representation was anyway. In fact `\d` in the pattern will produce "Nd"in the dump output.

Reply:
The bug was that these longer sequences were left in this the table, messing up the offsets for the "real" entries.